packages/shared-kernel/src/Result.ts

```diff
@@ -10,7 +10,7 @@ import { Narrow } from './ts-helpers/types';
  * @sealed
  * @privateRemarks DO NOT EXPORT, see type export later on
  */
-class Success<T, E> {
+class Success<T, E extends IEquatableError> {
```

Comment:
If we can afford the breaking change and removal of the second generic would simplify to `Success<T>` and `Failure<E>`. In this case you dont' even need `forward` as from the type perspective it should match the value correctly.

Reply:
I think it breaks the relationship with the underlying error/data?
<img width="393" alt="image" src="https://github.com/lens-protocol/lens-sdk/assets/3353250/6ca8b817-9d94-4b9a-8f0f-994b541639bd">

Reply:
even worse I get this:
<img width="480" alt="image" src="https://github.com/lens-protocol/lens-sdk/assets/3353250/9d41cdce-9bc7-4c14-80f5-4ead094554b7">

Reply:
Ignore me.